backend/dal/Services/Admin/Concrete/BaseService`.cs

```diff
@@ -0,0 +1,125 @@
+using System;
+using System.Security.Claims;
+using dal.Exceptions;
+using Pims.Dal.Entities;
+using Pims.Dal.Helpers.Extensions;
+
+namespace Pims.Dal.Services.Admin
+{
+    /// <summary>
+    /// BaseService abstract class, provides a generic service layer to perform CRUD operations on the datasource.
+    /// </summary>
+    /// <typeparam name="ET"></typeparam>
+    public abstract class BaseService<ET> : Pims.Dal.Services.BaseService<ET>, IBaseService<ET> where ET : BaseEntity
+    {
+        #region Constructors
+        /// <summary>
+        /// Creates a new instance of a BaseService class, and initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="dbContext"></param>
+        /// <param name="user"></param>
+        public BaseService (PimsContext dbContext, ClaimsPrincipal user) : base (dbContext, user) { }
+        #endregion
+
+        #region Methods
+        /// <summary>
+        /// Find the entity within the datasource with the specified key values.
+        /// </summary>
+        /// <param name="key"></param>
+        /// <returns></returns>
+        public virtual ET Find (params object[] keyValues)
+        {
+            // TODO: Check for system-administrator role.
+            if (this.User == null) throw new NotAuthorizedException ();
+
+            return this.Context.Set<ET> ().Find (keyValues);
+        }
+
+        /// <summary>
+        /// Add the specified entity to the datasource.
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        public virtual ET Add (ET entity)
+        {
+            this.AddOne (entity);
+            this.Context.CommitTransaction ();
+
+            return entity;
+        }
+
+        /// <summary>
+        /// Add the specified entity to the in-memory collection but do not commit to the datasource.
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        public virtual ET AddOne (ET entity)
```

Comment:
The [action]One functions provide a way to update the in-memory EF context but not push the change to the datasource immediately.  This is useful when performing multiple actions that should be part of a single transaction.